src/Traits/ArrayTrait.php

```diff
@@ -30,6 +32,30 @@ protected function getFlatArray(array $array, string $delimiter, string $rootPre
         return $result;
     }
 
+    /**
+     * @param string[] $array
+     * @param Filter $filter
+     *
+     * @return array<int, string[]>
+     */
+    protected function getFilteredResult(array $array, Filter $filter): array
+    {
+        $translations = [];
+        $filteredTranslations = [];
+
+        foreach ($array as $key => $value) {
+            if ($filter->isKeyAllowed($key, true) === true) {
```

Comment:
the only thing that i dont really get is the partial allowing?
why do we have that?
can you also explain that in a comment.
if i want to filter an array, i dont see a point where i would need partial filtering

Reply:
Before I add the comment I explain it here, maybe there is a better way for the whole solution.

Background is that filtering during the first run through of flattening the array was no good solution in my opinion. During implementation there were too many changes needed to be done to save the filtered results. Therefore I moved it to this new function. As this runs after the array is flattened we have an array like this:
```
$array = [
    'app.name' => 'test',
    'app.description' => 'test',
];
```
If we now want to filter out description, we cannot filter by an exact match as the string would is `app.description` therefore we need the partial match.